apps/cow-fi/app/(learn)/learn/articles/[[...pageIndex]]/page.tsx

```diff
@@ -18,32 +17,38 @@ export type ArticlesResponse = {
 }
 
 export async function generateStaticParams() {
-  const articlesResponse = await getArticles({ page: 0, pageSize: ITEMS_PER_PAGE })
+  const articlesResponse = await getArticles({ page: 0, pageSize: ARTICLES_PER_PAGE })
   const totalArticles = articlesResponse.meta?.pagination?.total || 0
-  const totalPages = Math.ceil(totalArticles / ITEMS_PER_PAGE)
+  const totalPages = Math.ceil(totalArticles / ARTICLES_PER_PAGE)
 
   return Array.from({ length: totalPages }, (_, i) => ({ pageIndex: [(i + 1).toString()] }))
 }
 
 export default async function Page({ params }: Props) {
-  const pageParam = (await params)?.pageIndex
-  const paramsAreSet = Boolean(pageParam && pageParam.length > 0)
+  const pageParam = (await params)?.pageIndex?.[0]
+  const paramsAreSet = Boolean(pageParam)
   const pageIndexIsValid = Boolean(pageParam && /^\d+$/.test(pageParam))
 
+  // Redirect invalid page numbers to the main articles page
   if (paramsAreSet && !pageIndexIsValid) {
     return redirect('/learn/articles')
   }
 
   const page = pageParam && pageIndexIsValid ? parseInt(pageParam, 10) : 1
 
   // Fetch paginated articles for display
-  const articlesResponse = (await getArticles({ page, pageSize: ITEMS_PER_PAGE })) as ArticlesResponse
+  const articlesResponse = (await getArticles({ page, pageSize: ARTICLES_PER_PAGE })) as ArticlesResponse
+  const totalArticles = articlesResponse.meta?.pagination?.total || 0
```

Comment:
it was hear already, but I would think the casting is not needed

Reply:
Addressed in https://github.com/cowprotocol/cowswap/pull/5655 and merged in this release branch.